library/alloc/src/collections/linked_list.rs

```diff
@@ -639,7 +679,7 @@ impl<T> LinkedList<T> {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn clear(&mut self) {
-        *self = Self::new();
+        while self.pop_front_node().is_some() {}
```

Comment:
I haven't looked at the Drop impl, but this seems less than optimal - previously we were presumably dropping the T in place rather than copying (moving it) out and then dropping it.

Note that this is really caused by the change to the pop function returning Node rather than the Box of a Node.

Reply:
This also means that we're losing the panic "continue dropping" behavior of the drop impl.

Reply:
Good point; I'll see if there's a way to avoid the excess copies.

Would it be reasonable to move the node drop code into `clear` and then have the drop impl call `clear` so they both get the panic guard?

Reply:
Seems plausible.